pkg/controllers/resourceapply/resourceapply.go

```diff
@@ -129,13 +129,17 @@ func applyConfigMap(ctx context.Context, client coreclientv1.Client, recorder re
 
 func applyDeployment(ctx context.Context, client coreclientv1.Client, recorder record.EventRecorder, requiredOriginal *appsv1.Deployment) (bool, error) {
 	required := requiredOriginal.DeepCopy()
-	err := SetSpecHashAnnotation(&required.ObjectMeta, required.Spec)
-	if err != nil {
+	if err := annotatePodSpecWithRelatedConfigsHash(ctx, client, required.Namespace, &required.Spec.Template); err != nil {
+		klog.V(5).Infof("Can not check related configs and secrets: %w", err)
+		recorder.Event(required, corev1.EventTypeWarning, "Can not check related configs", err.Error())
```

Comment:
when making events, the [general convention](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#events) for the "reason" is to use a single CamelCase word, and then use something more free form for the "message" part.  in this case we might do something like:

```suggestion
		recorder.Event(required, corev1.EventTypeWarning, "FailedConfigurationCheck", fmt.Sprintf("Can not check related configs, error: %v", err.Error()))
```

and similar on the other events too

Reply:
Good suggestion. Replaced all reasons in this file with constants, added that as separate commit: https://github.com/openshift/cluster-cloud-controller-manager-operator/pull/227/commits/c2fdaa97c429ab2f05769a4bca20b1b287dcb4ca